assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
+	useRunOnEscape( onClose );
+
+	return (
+		<div className={ 'sensei-modal' }>
+			<button
+				className="sensei-modal sensei-modal__overlay"
+				aria-label="Close"
```

Comment:
We need to add the `__` here to make it translatable.

Reply:
Oops! Now that I noticed it's the overall. I think it's not very good from an accessibility perspective. Maybe we could try to use the same as Gutenberg modal? The `__experimentalUseFocusOutside`.

Reply:
Reimplemented modal using your suggestions here: 0c029ec98